datasource/netcdf_groundwater.py

```diff
@@ -280,121 +277,115 @@ def get_gridadmin(self, variable=None):
-            else:
-                a1d = np.ma.masked_all(n1d)
-
-            assert var_2d in ds.keys() or var_1d in ds.keys(), "No 2D and 1D?"
-
-            # Note: order is: 2D, then 1D
-            res = np.ma.hstack([a2d, a1d])
+            return filtered_data
+
+    def _nc_from_mem(self, variable):
+        """Return 2d numpy array with all values of variable and cache it.
+
+        Everyting of the variables is cached, both in time and space, i.e. all
+        timesteps and all nodes of the variable.
+
+        TODO: Saving the variables in cache is currently necessary to limit
+         the amount of (slow) IO with the netcdf results. However, this also
+         causes many unnecessary values to be stored in memory. This can become
+         problematic with large result files.
+
+        :param variable: (str) variable name, e.g. 's1', 'q_pump'
+        :param use_cache: bool
+        :return: 2d numpy array
+        """
+        if variable in self._cache:
+            values = self._cache[variable]
         else:
-            # in the groundwater version, the node index starts from 1 instead
-            # of 0.
-            # Note: a new array is created, e.g., index doesn't get modified
-            index = index - 1  # copies the array
-
-            # hacky object_type checking mechanism, sinds we don't have
-            # that information readily available
-            if variable.startswith("q_pump"):
-                return self._nc_from_mem(ds, var_1d, use_cache)[timestamp_idx, index]
-            elif variable in ALL_Q_TYPES:
-                threshold = self.nMesh2D_lines
-            elif variable in ALL_H_TYPES:
-                threshold = self.nMesh2D_nodes
-            else:
-                raise ValueError(variable)
-            # find indices of 2d and 1d components
-            idx_2d = np.where(index < threshold)[0]
-            idx_1d = np.where(index >= threshold)[0]
-            # make index arrays that can be used on the nc variables
-            iarr_2d = index[idx_2d]
-            iarr_1d = index[idx_1d] - threshold
-            res = np.ma.zeros(index.shape, fill_value=fill_value)
-            # Note sure if a netCDF bug or a known difference in behavior.
-            # Indexing a numpy array using [], or np.array([], dtype=int)
-            # works, but on a netCDF Variable it doesn't. Therefore we must
-            # explicitly check if the list is empty.
-            if iarr_2d.size > 0:
-                res[idx_2d] = self._nc_from_mem(ds, var_2d, use_cache)[
-                    timestamp_idx, iarr_2d
-                ]
-            if iarr_1d.size > 0:
-                res[idx_1d] = self._nc_from_mem(ds, var_1d, use_cache)[
-                    timestamp_idx, iarr_1d
-                ]
-        # note: res is a masked array
-        return res
+            logger.debug(
+                "Variable %s not yet in cache, fetching from result file" % variable)
```

Comment:
Die % moet je eigenlijk niet gebruiken in logger.nogwattus()
https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html